### PR TITLE
[Demo] Add diverging bar to visual-vocabulary

### PR DESCRIPTION
**Pull request by hxe00570**
## Description
- Add diverging bar to visual-vocabulary

## Screenshot
Screenshot 2024-10-10 at 15 01 29

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

**Comment by nadijagraca** [comment]
Hi @hxe00570, thank you for contributing. 🙌 

There are several things that need to be added to this PR for it to be merged. 
I've summarised these things in the to-do list below:

- [ ] create complete [custom_chart](https://vizro.readthedocs.io/en/stable/pages/user-guides/custom-charts/) example for diverging bar chart and add it to a page titled "Diverging bar". 
- [ ] update README.md file
- [ ] Add changelog file
- [ ] Add chart to relevant category, also remove the chart from incomplete charts. 
- [ ] update PR description with chart description and screenshot 

Also, you could try check and fix the 3 failed CI? 
- Checks for Vizro / checks-vizro-core (pull_request)
- Lint / lint-vizro-all (pull_request)
- pre-commit.ci - pr

Can you please advise if you wish to continue working on this issue, or some other contributor can take it from here? 

Thank you, 🙏

**Comment by nadijagraca** [comment]
> Thank you, I think I corrected this. Please inform

Can you please push your latest changes as I cannot see new commits. 

Thank you,

**Comment by lingyielia** [comment]
Hi @hxe00570 , we are not participating GHC in person this time. I have DMed you via Linkedin in case you'd like to chat. 😄 

In the meantime, I saw another PR from you also related to diverging bar (https://github.com/mckinsey/vizro/pull/795). If they are duplicate, we can close one and keep working on one git branch.

**Comment by huong-li-nguyen** [comment]
> I have tried a few times to push and I keep getting different errors (this is my first contribution so I’m sure it’s user error). I have been using the Codespace. Any chance yall are on site at GHC and someone might be willing to meet up and help me?

Hello @hxe00570 ,

I just looked at the chart you created in this PR: https://github.com/mckinsey/vizro/pull/795 and it looks beautiful!  ⭐ I also noticed that you tackled the more challenging task of creating a diverging stacked bar chart! 🚀

If you find yourself short on time or encountering issues with the development setup, we're happy to help you wrap this up within your branch. This way, you'll still receive credit for the contribution, and we can handle any remaining technical details. Please let me know if you'd prefer to complete this on your own or if you'd like us to assist in finishing it up within your branch :)

**Comment by lingyielia** [comment]
@nadijagraca @huong-li-nguyen Had a quick chat with @hxe00570 today and she is ready to handover the pr to Vizro team now. 

I will close this PR (#792) and we can assist in finishing it up in PR https://github.com/mckinsey/vizro/pull/795. 🙌

**Comment by huong-li-nguyen** [comment]
Reopening this PR because @hxe00570 actually worked on the diverging stacked bar chart, so I'll use the two PRs to finish off 2 charts: the diverging bar chart and diverging stacked bar chart :)

**Comment by huong-li-nguyen** [comment]
@hxe00570  - I really liked your pastry example, so I've kept the data but I just replaced the measure with the profit ratio (fits better with negative and positive values) :)

You tackled the more complex chart which is the diverging stacked bar chart - so don't worry I'll re-use the code you created in another PR: https://github.com/mckinsey/vizro/pull/795

For the diverging bar chart, we were actually just looking for a simple sorted bar chart. So, in this PR, I am simplifying the function call by using the pure `px.bar` call 👍

**Review by antonymilne — APPROVED** [review]
Thank you for all your work on this @hxe00570! 🙏 

@huong-li-nguyen I think there's something small we should change here - see comment. But otherwise looks good! :shipit:

File: vizro-core/examples/visual-vocabulary/pages/deviation.py [hunk 1]
```diff
@@ -1,7 +1,53 @@
...
+        """
+        ),
+        vm.Graph(
+            figure=px.bar(
+                pastries.sort_values("Profit Ratio"),
+                orientation="h",
+                x="Profit Ratio",
+                y="pastry",
+                color="Profit Ratio",
+                color_continuous_scale=COLORS["DIVERGING_RED_CYAN"],
```
[the remark this answers is not in the document]
Comment (huong-li-nguyen) [reply vizro-core/examples/visual-vocabulary/pages/deviation.py]: Ahh, that's good to know 👍

**Review by nadijagraca — APPROVED** [review]
The chart looks great, well done @hxe00570  and @huong-li-nguyen . 🚀

**Pull request merged**